version: '3.6'
  
services:
  core:
    image: qbonaventure/discord-bot-core
    tty: true
    restart: always
    links:
      - core-db
      - core-cache
    env_file:
      - .env
    volumes:
      - core_composer_cache:/app/vendor
    deploy:
      labels :
        ftcbot.service: core
      replicas: 4
      update_config:
        parallelism: 2
        delay: 20s
      restart_policy:
        condition: on-failure
    networks:
      ftcbot:
#    command: sh /app/entrypoint.sh
      
  core-db:
    image: postgres:10-alpine
    expose:
      - "5432"
    deploy:
      labels:
        ftcbot.service: core-db
      replicas: 1
      placement:
        constraints:
          - node.role == manager
    environment:
      POSTGRES_PASSWORD: $FTCBOT_DB_PASSWORD
      POSTGRES_DB: $FTCBOT_DB_DBNAME
    volumes:
      - core_db:/var/lib/postgresql/data:z
    networks:
      ftcbot:
      
  core-cache:
    image: redis:alpine
    expose:
      - "6379"
    volumes:
      -  core_cache_data:/data
    deploy:
      labels:
        ftcbot.service: core-cache
      replicas: 1
      placement:
        constraints:
          - node.role == manager
      restart_policy:
        condition: on-failure
    networks:
      ftcbot:

networks:
  ftcbot:

volumes:
  core_db:
  core_composer_cache:
  core_cache_data: 
